android/src/main/java/com/swmansion/rnscreens/ScreenContainer.kt

```diff
@@ -103,6 +103,13 @@ open class ScreenContainer<T : ScreenFragment>(context: Context?) : ViewGroup(co
     open fun removeScreenAt(index: Int) {
         mScreenFragments[index].screen.container = null
         mScreenFragments.removeAt(index)
+
+        // If screen at given index had another screen before in stack,
+        // calculate its header height.
```

Comment:
We need this, because when there's a screen that is exiting to its parent it seems that it doesn't call `calculateHeaderHeight` (You can check it on video below, where I'm testing it in ScreensExample). However, it looks like this is not the case in TestsExample, so I assume this is necessary for the nested navigators, when we're traversing from child navigator to the parent one? 🤔 



Also, I think caring about triggering this method in other navigators is not necessary - shouldn't this method be called only for ScreenContainer that contains only screen with given index? Otherwise calls from lines 104-105 would throw an exception there because of mScreenFragments[index] 😄

Reply:
I am not sure why should we trigger this event when going back to a screen? Could you elaborate on this more? I thought that each screen has its own header height measurement since each of them can have a header. So when going back to a screen, why would its header change from the time we left it?

Reply:
@WoLewicki I deeply tested that and it looks like currently NativeStackView loses the information about previous header height 😕 

https://github.com/software-mansion/react-native-screens/assets/23281839/40a388da-7cfa-475f-8ebf-d38506b10cb3

I assume that's because `Animated` does not hold the information between renders. I managed to solve that by wrapping `Animated` with useRef, but to be honest I'm not convinced if doing `React.useRef().current` and performing operations on it as usual is a good solution (and even practice?). What do you think? 🤔

Reply:
As I considered it, I don't see any points on which circumstances it might not work (I don't see any pitfalls either), so I assume we might go with that 🤔

Reply:
I've removed the android part of the code and changed the JS side here: https://github.com/software-mansion/react-native-screens/pull/1802/commits/2edab997027f41c73cc9422c085170bce290bcf2.